ngraph/frontend/paddlepaddle/src/op/range.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "range.hpp"
+#include <ngraph/opsets/opset6.hpp>
+
+namespace ngraph
+{
+    namespace frontend
+    {
+        namespace pdpd
+        {
+            namespace op
+            {
+                NamedOutputs range(const NodeContext& node)
+                {
+                    auto start = node.get_ng_input("Start");
+                    auto stop = node.get_ng_input("End");
+                    auto step = node.get_ng_input("Step");
+
+                    bool keep_dims = false;
+                    const auto axis = ngraph::opset6::Constant::create(element::i64, Shape{}, {0});
+                    auto start_scalar =
+                        std::make_shared<ngraph::opset6::ReduceMin>(start, axis, keep_dims);
+                    auto stop_scalar =
+                        std::make_shared<ngraph::opset6::ReduceMin>(stop, axis, keep_dims);
+                    auto step_scalar =
+                        std::make_shared<ngraph::opset6::ReduceMin>(step, axis, keep_dims);
+
+                    // TODO to support other data types other than FP32 #55267
+                    return node.default_single_output_mapping(
+                        {std::make_shared<ngraph::opset6::Range>(
+                            start_scalar, stop_scalar, step_scalar, element::f32)},
```

Comment:
We specifically added `get_out_port_type` function to `NodeContext` to support different types. Please use node.get_out_port_type("Out") to get output type.

Reply:
@mangguo321 Could check this part ?